tests/org.dataflowanalysis.analysis.tests/src/org/dataflowanalysis/analysis/tests/dfd/OnlineShopDFDCyclicTest.java

```diff
@@ -0,0 +1,151 @@
+package org.dataflowanalysis.analysis.tests.dfd;
+
+import static org.dataflowanalysis.analysis.tests.AnalysisUtils.TEST_MODEL_PROJECT_NAME;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+
+import java.nio.file.Paths;
+import java.util.List;
+import org.dataflowanalysis.analysis.core.AbstractVertex;
+import org.dataflowanalysis.analysis.core.DataCharacteristic;
+import org.dataflowanalysis.analysis.dfd.DFDConfidentialityAnalysis;
+import org.dataflowanalysis.analysis.dfd.DFDDataFlowAnalysisBuilder;
+import org.dataflowanalysis.analysis.dfd.core.DFDCharacteristicValue;
+import org.dataflowanalysis.analysis.dfd.core.DFDFlowGraphCollection;
+import org.dataflowanalysis.analysis.dfd.core.DFDVertex;
+import org.dataflowanalysis.examplemodels.Activator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class OnlineShopDFDCyclicTest {
+
+    private DFDConfidentialityAnalysis analysis;
+
+    @BeforeEach
+    public void initAnalysis() {
+        final var dataFlowDiagramPath = Paths.get("models", "OnlineShopDFD", "onlineshop.dataflowdiagram");
+        final var dataDictionaryPath = Paths.get("models", "OnlineShopDFD", "onlineshop.datadictionary");
+
```

Comment:
How about a test case that runs the analysis once with the normal finder and once with the cyclic finder and assures both flowgraph collections are the same. Or would that be an overkill?

Reply:
the problem with that is that you would need to provide the transposeFlowGraphFinder with the a resource provider, and i did not take a look into this process. Also this should be given, since those are the same parameters as the normal test.